test/e2e/util.go

```diff
@@ -3753,3 +3767,29 @@ func CoreDump(dir string) {
 		Logf("Error running cluster/log-dump.sh: %v", err)
 	}
 }
+
+// Run a function over each pod in a namespace.
+// Meant to be called in a loop - where the fn does some assertion logic on a test (i.e. look for string in the pod logs)
+func TryEachPod(c *client.Client, ns, selectorKey, selectorValue string, fn func(api.Pod)) {
+	pods := []api.Pod{}
+	for t := time.Now(); time.Since(t) < podListTimeout; time.Sleep(poll) {
+		selector := labels.SelectorFromSet(labels.Set(map[string]string{selectorKey: selectorValue}))
+		options := api.ListOptions{LabelSelector: selector}
+		podList, err := c.Pods(ns).List(options)
+		Expect(err).NotTo(HaveOccurred())
+		for _, pod := range podList.Items {
+			if pod.Status.Phase == api.PodPending || pod.Status.Phase == api.PodRunning || pod.Status.Phase == api.PodSucceeded {
```

Comment:
This or logic on pods should probably be a separate func, or applied as the fn() operation b/c we're pre-filtering the list.